proposals/p1871.md

```diff
@@ -0,0 +1,104 @@
+# Introduce `var`/`let`-`else`-statement
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/1871)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+-   [Unresolved questions](#unresolved-questions)
+
+<!-- tocstop -->
+
+## Problem
+
+This statement is syntactic sugar for a set of other patterns, specifically it
+guards a code block if a requirement of this code block is violated. That is,
+there is actually no problem.
+
+## Background
+
+The idea goes back to the prior art in
+[Swift](https://docs.swift.org/swift-book/LanguageGuide/ControlFlow.html#ID525)
+to allow the execution of a statement depending on the Boolean value of an
+expression. Swift used a **guard** statement to require that a condition must be
+true in order for all following code in the current code block to be executed.
+If the condition is not true, control needs to be transferred using a control
+transfer statement as `break`, `continue`, `return`, or a function that does not
+return.
+
+[Rust](https://github.com/rust-lang/rfcs/blob/master/text/3137-let-else.md)
```

Comment:
Can you instead add the actual [pull](https://github.com/rust-lang/rfcs/pull/3137) so that our reviewers can feel the impression at that time?